### DISCUSSION TITLE
[Q/A]How does provider response attachments to consumer in Dubbo2.7.18?

### DISCUSSION THREAD

**Discussion by LiebeKrake** [apache/dubbo#13924] [Question]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

provider：
```
public class MyGenericService implements GenericService {
public Object $invoke(String methodName, String[] parameterTypes, Object[] args) throws GenericException {
RpcContext serverContext = RpcContext.getServerContext();
serverContext.setAttachment("key","value");
return "xxxx";
}
```

consumer:
```
registryConfig.setAddress("N/A");
...
referenceConfig.setUrl("dubbo://localhost:8888");
referenceConfig.get().$invoke("echo", new String[]{"java.lang.String"}, new String[]{body});
RpcContext.getServerContext().getAttachment("key");
```

——————
My code snippet is shown above,
I cannot get the value of the attachment "key" on the consumer side. How should I return attachments from the provider to the consumer?
(For some special reasons, only dubbo2 can be used for the time being)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
The usage is OK and this might be a bug of 2.7.x. But 2.7.x is EOL now, we will not release a new version for 2.7.x.

Here are some advice for you if you want to know why.
1. Check `org.apache.dubbo.rpc.filter.ContextFilter#onResponse` in provider side whether set the attachments into `appResponse`
2. Check `org.apache.dubbo.rpc.cluster.interceptor.ConsumerContextClusterInterceptor#onMessage` in consumer side whether the `appResponse` contains the attachments and set to `ServerContext` success or not
